pkg/query-service/app/logs/v4/resource_query_builder.go

```diff
@@ -0,0 +1,204 @@
+package v4
+
+import (
+	"fmt"
+	"strings"
+
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+	"go.signoz.io/signoz/pkg/query-service/utils"
+)
+
+// buildResourceFilter builds a clickhouse filter string for resource labels
+func buildResourceFilter(logsOp string, key string, op v3.FilterOperator, value interface{}) string {
+	// we are using lower(labels) as we want case insensitive filtering
+	searchKey := fmt.Sprintf("simpleJSONExtractString(lower(labels), '%s')", key)
+
+	chFmtVal := utils.ClickHouseFormattedValue(value)
+
+	switch op {
+	case v3.FilterOperatorExists:
+		return fmt.Sprintf("simpleJSONHas(lower(labels), '%s')", key)
+	case v3.FilterOperatorNotExists:
+		return fmt.Sprintf("not simpleJSONHas(lower(labels), '%s')", key)
+	case v3.FilterOperatorRegex, v3.FilterOperatorNotRegex:
+		return fmt.Sprintf(logsOp, searchKey, chFmtVal)
+	case v3.FilterOperatorContains, v3.FilterOperatorNotContains:
+		// this is required as clickhouseFormattedValue add's quotes to the string
+		lowerEscapedStringValue := utils.QuoteEscapedString(strings.ToLower(fmt.Sprintf("%s", value)))
+		return fmt.Sprintf("%s %s '%%%s%%'", searchKey, logsOp, lowerEscapedStringValue)
+	default:
+		chFmtValLower := strings.ToLower(chFmtVal)
```

Comment:
On the main table also do we follow the case-insensitive search?

Reply:
For tag attributes, it's still case sensitive  due to limitations on map indexes.

All resources filtering goes to this table so all resource attributes search is case insensitive.

Reply:
I don't this is a good idea overall i.e search on resource attributes is case insensitive but a search on regular attributes is case sensitive. The filter by resource is common so the user forms an impression that a search on any attributes is case-insensitive but then when they do search on attributes with the same expectation it will fail.

Reply:
So we want to get to a place we it's insensitive for attributes as well, but the limitation is restricting us. We eventually want to pick it up.

Reply:
When is that eventually thing going to happen? It wouldn't be a problem if the feature were to be rarely used. but working with attributes happens all the time and people shouldn't learn it hard way that resource attributes and regular attributes work differently.

Reply:
> When is that eventually thing going to happen
I don't have time that I can give you.

I agree that there will be a different curve for this, but @ankitnayan  suggested going ahead with this as a starting point.